api/src/main/java/org/pentaho/platform/api/genericfile/GetTreeOptions.java

```diff
@@ -34,6 +34,15 @@ public class GetTreeOptions {
   @Nullable
   private GenericFilePath expandedPath;
 
+  /**
+   * Enum to represent the three filters for 
+   */
+  public enum TreeFilter {
+    FOLDERS, FILES, ALL;
+  }
+
+  private TreeFilter treeFilter = TreeFilter.ALL;
```

Comment:
Field and property could be called `filter` and `getFilter`/`setFilter` directly. The `Tree` word is implied by the name of the class, `GetTreeOptions`.